### PR TITLE
Add support for Backend-JWT

### PR DESCRIPTION
**Pull request by shanakama**
## Related Issues
https://github.com/wso2-enterprise/choreo/issues/8542

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.common/src/main/java/org/wso2/carbon/apimgt/rest/api/common/RestAPIAuthenticationManager.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class RestAPIAuthenticationManager {
+
+    private static final Log log = LogFactory.getLog(RestAPIAuthenticationManager.class);
+    private static RestAPIAuthenticator authenticator = null;
+
+    public static RestAPIAuthenticator getAuthenticator() {
+        if (authenticator == null) {
+            ServiceReferenceHolder serviceReferenceHolder = ServiceReferenceHolder.getInstance();
+            if (serviceReferenceHolder.getAuthenticator() != null) {
+                log.info("Authenticating in Back-end JWT");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.common/src/main/java/org/wso2/carbon/apimgt/rest/api/common/RestAPIAuthenticationManager.java:32]: use debug

Comment (shanakama) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.common/src/main/java/org/wso2/carbon/apimgt/rest/api/common/RestAPIAuthenticationManager.java]: Fix with : https://github.com/wso2/carbon-apimgt/pull/10870/commits/7c1b7429c987ede0b4c4f7fe5fca870b33ef7d25

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java [hunk 2]
```diff
@@ -58,48 +61,87 @@ public OAuthAuthenticationInterceptor() {
...
+                        logger.info("Request with the Backend JWT has been Authenticated");
+                    } else {
+                        logger.error("Request based on Backend JWT is not Authenticated");
+                        throw new AuthenticationException("Unauthenticated request");
+                    }
+                } catch (APIManagementException e) {
+                    logger.error("Backend JWT Authentication Failure");
+                    return;
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java:116]: remove

Comment (shanakama) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java]: Fix with : https://github.com/wso2/carbon-apimgt/pull/10870/commits/637afe2b0aeb43831584d3ca2ba30528c35e37c4

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**